.github/workflows/build.yml

```diff
@@ -47,6 +47,28 @@ jobs:
           - "3.11"
           - "3.12"
           - "3.13"
+        exclude:
+          # importing the package from the wheel instantly segfaults on macOS Python <= 3.12
```

Comment:
@zacharyburnett Have you found out why that's the case? I have been trying to build a wheel for macos arm64 for python 3.10. I can build the  wheel and install it and run it (`import s2geometry`) on any environment that shares the same base python. Eg venvs created using homebrew's 3.10.  But as soon as I switch to another installation (eg pyenv 3.10) I get segfaults. Something about KERN_INVALID_ADDRESS.

I used delocate-wheel to "fix" the wheel so it includes the necessary libraries as dylib. The resulting wheel segfaults even if the same base python was used. I noticed that Python.dylib is also linked and included.

Reply:
unfortunately, this is the limit of my knowledge of libraries and compilation 😓 

At the very least, we have working Python 3.13 wheels on macOS

Reply:
Does this mean you are able to run a delocated wheel built with 3.13?

Reply:
I didn't even have to `delocate` it, it just works 🤷

Reply:
How did you test it works? Do you have a wheel you can share?

Reply:
sure, here're the tests on `macos-13` (Intel) and `macos-latest` (ARM64): https://github.com/zacharyburnett/s2geometry/actions/runs/11935111543/job/33265728139

The tests fail with assert differences, but they actually run instead of segfaulting

Here're the built wheels: https://github.com/zacharyburnett/s2geometry/actions/runs/11935111540

Reply:
The whole problem is being able to install the wheel on another machine with similar architecture. I have only been able to share a wheel I built by instructing them to only use python 3.10 and abseil from homebrew. (btw abseil from homebrew also works).

The above wheel will fail on my machine if try it on an environment built using pyenv's python.

Reply:
Aha, you're right! In a new environment on a macOS Intel machine I get `Library not loaded`:
```
micromamba create -n s2py313 python=3.13 pytest -y
micromamba activate s2py313
pip install ~/Downloads/s2geometry-0.0.0-cp313-cp313-macosx_13_0_universal2.whl
pytest src/python/s2geometry_test.py
```
```
...
E     Reason: tried: '/Library/Frameworks/Python.framework/Versions/3.13/Python' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Library/Frameworks/Python.framework/Versions/3.13/Python' (no such file), '/Library/Frameworks/Python.framework/Versions/3.13/Python' (no such file)
```

Reply:
oof, it gets a segfault when I `delocate-wheel` and then try running that on my machine, just as you said 😓